### DISCUSSION TITLE
Getting going - looking for Circuits / features and Aux Circuits explanation

### DISCUSSION THREAD

**Discussion by bwinstonbillville** [tagyoureit/nodejs-poolController#990] [Q&A] [answered]
Thanks for all the work Ya'll do - this is awesome.

Have hardware working - I can switch my relays and valves on the test bench.  Trying to setup dashPanel to do what I want and then schedule it.  I have identified 4 things I want to put on the schedule and I cannot find the explantion of general theory of how PoolController / dashPanel operates.  I'm needing to translate the 4 modes below with the 4 valve positons of each to how this software operates.  I'm willing to do the reading - really just looking to where the explanation is - I can't seem to find it in the Wiki. 

Pool Mode Jets / Skim	
Intake off to spa	
i2 down off to pool drain	
Return off to spa	
r2 off to Herbert	
	
Pool Mode Herbert	
Intake off to spa	
i2 down off to no selection	
Return off to spa	
r2 off to Pool Jets	
	
Spa Mode Heat	
intake off to skim	
I2 off to pool drain	
return off to pool	
r2 not relevant	
	
Spa Mode Overflow	
intake spa drain off	
i2 down to no selection	
Return off to pool	
r2 not relevant

**Comment by bwinstonbillville** [comment]
This stuff is just incredible.

So...  I'm setting up poolController solo without an Intellicenter and getting the idea that is not as popular a setup as with an Intellicenter.  All good, but I suspect the configuration is just a bit different.

I have setup my 4 Valves and pump (single speed) in Aux Circuits, created Circuit Groups for my 4 modes in the previous post with the 4 valves and the pump.  Then set the schedule using the Circuit Groups.

On the test bench it works great.  I am not sure what is normal as I have never had a Pool before and therefore never even seen an Intellicenter.

I'm excited that it works the way I was hoping - I've no idea what is normal or if this is how others using this standalone (no intellicenter) do it. 

My valves are turning, my 3HP is clicking - VERY Excited - did a little happy dance.  Now to figure out the heater.

**Comment by rstrouse** [comment, marked as answer]
There are 4 main valve modes for as shared body pool/spa.  These include

Pool Mode:
Intake: Not diverted
Return: Not diverted

Spa Mode
Intake: Diverted
Return: Diverted

Spillway: (this is actually a pool mode but the return is to the spa so it overflows)
Intake: Not Diverted
Return: Diverted

Spa Drain:
Intake: Diverted
Return: Not Diverted

Essentially the Intake/Return valves should control all of the flow for the pool and spa modes.  These two valves control where the water is drawn from and returned to.  However, it is not uncommon to have a 3rd valve on the intake side for things like a suction side cleaner or a skimmer.  And another for spa jets downstream of the spa return.

Most folks do not switch between the main drain and the skimmer and simply balance the this on the intake side but Nixie will support you adding additional valves to turn off your skimmers.  Keep in mind when you are in spa mode you are isolating the plumbing from the pool and pool mode should be the default undiverted positions of the valves.  This means that when neither the pool or spa are selected it should default to both intake and return in an undiverted state.   Intake from the pool and return to the pool.

**Comment by bwinstonbillville** [comment]
Sorry was on vacation for a week.  Also most of your answers are succinctly and precisely stated - but it takes me days to read, reread, fully comprehend and implement your answers.  Not a knock on you at all,  just takes a minute.  

I have all the valve actuators now and will be hooking them up in the system and seeing if I can successfully schedule all four modes.  Hoping to have it in by this weekend.  Will post up a pic if I get it working.  

How do we donate to the cause?  Certainly an enormous effort has gone into this venture.  Thanks for all you guys do and your patience.

**Reply by rstrouse** [reply]
Keep in mind that valves are driven by the current state of the circuits and features in the pool.   So when the system is in pool mode the intake/return valves will be in the correct position.   If it is in pool mode and a spillway feature is activated it will reposition the valves to spill over the spa.   If it is in spa mode it will reposition the valves to isolate the spa and if a drain feature is activated it will reposition the valves to drain the spa.   All you need to do is hook up the valves to the relays and the programming in njsPC will do the rest.

For accessory valves (those that are not intake or return) simply assign a circuit or feature to them and it will divert when that circuit or feature is activated.  This includes virtual circuits (those that you do not manually control) like solar.  Solar is activated when the temperature falls below the set temp and the conditions for engaging the solar are met (body on, differential met, .... etc).

**Reply by bwinstonbillville** [reply]
thanks so much for that info.  I was trying to over do it as usual.  Removed all the extra aux circuits and features and just went with default pool, spa and added the spill over and works great on the bench.  So I can pool mode, spa mode and then turn on the spa overflow and the valves turn as expected.

My setup may not be normal.  I don't know as this is my first pool.  This is how my valve setup is arranged.  Quick diagram but I think it will work.
![pool valve diagram](https://github.com/user-attachments/assets/46f889cf-99d0-4ae0-b1b2-c847f2ba7e31)

**My question is specifically on the Intake side and Intake 2 valve.  In SPA mode the Intake valve would be diverted (Off marking on the handle pointing down) so when in SPA mode the Intake 2 Valve would need to close off the Pool drain.**  

In POOL mode the intake valve would be not diverted (off marking on the handle pointing  to the spa jets) and **I need the Intake 2 valve to switch to not diverted (off pointing down).**

So how do I setup Intake 2 to switch with Pool Mode and Spa mode?  Seems like I would add this aux circuit and set the type to Pool or Spa but that doesn't move I2 valve when I switch Pool to Spa.  Do I need to do Circuit groups for this?

I know people like pics so here is my box...
![automation box](https://github.com/user-attachments/assets/38979afd-55e9-4024-ba32-b15fae852cc0)

again, I can't thank you enough for your patience and help.  On the home stretch!
Still need to get my temp probes and heater going....  baby steps.

**Reply by rstrouse** [reply]
Are you certain that is how you are plumbed on the intake side.  Typically the intake chooses between pool or spa and the pool line is split between drain and skimmer.  Your plumbing on the intake side is very strange.   The return looks about right with the pressure side cleaner and the pool returns.

**Reply by bwinstonbillville** [reply]
Yes sir,  Here is a picture of the pool pad.
![pool pad](https://github.com/user-attachments/assets/318635d5-c850-4764-832e-51c9976007f9)

obviously the intake is on the right and you can see Intake 2 valve from the diagram where the pipe goes into the pump.  The Spa drain is on the far right.

**Reply by rstrouse** [reply]
Are you sure the valve on the far right does not switch between skimmer and bottom drain?

**Reply by bwinstonbillville** [reply]
100 percent.  I spend a lot of time with these valves as the system is fully manual.  

Full spa mode is intake side - off towards skimmer - off towards pool drain /  return off to the left 

this absolutely isolates the spa and recirculates spa water through the filter.  

Thanks again for spending time on this

**Reply by bwinstonbillville** [reply]
I could re plumb easy enough - couple valves and some pipes, couplers, Couple 45s etc.  not the end of the world really.

or for the mean time - I could just keep the pool drain off and use the main intake valve to switch between pool mode (skimmer) and spa mode.

love the pool - builder may have sucked at plumbing setup, but 2 hp pump and a 60 sq ft DE filter is perfectly over done for 6k gallons.  

When the pump dies I can get a nice VS pump and figure out the 485 lol

**Reply by rstrouse** [reply]
It is just very weird that the balance valve on the intake would balance between skimmer and spa.  Does the spillway spill from the spa to the pool or the other way around?

**Reply by bwinstonbillville** [reply]
Spillway spills from the spa to the pool.
![spillway](https://github.com/user-attachments/assets/f9537ddd-6c03-4016-bf2f-126aad42746d)

**Reply by bwinstonbillville** [reply]
ok so I'm good for right now moving I2 by selecting the feature - I've added it via Aux circuits and can manually engage when I want to.  Very pleased with valve functions.  I may later re-arrange the plumbing later, but not that big of a deal right now.

I've got the lights setup on a relay that switches 120v to my lights and that is working now.

I've go the fireman's switch setup on a relay that just clicks the NO to COM on a relay and wired up.  Seems to be working - I'll hook it up and find where it goes on my Hayward 400k gas heater.  Seems easy enough remove the jumper and replace with the wires from the relay.

**QUESTION:  How do I get the Temp probe for the pool temp to show up in the dashboard.**  The air temp is working (I have two probes)
- both show up in REM under inputs and show 10.9K and 10.6K ohms.  They are right next to each other.
- both have a feed setup using the in0_10x.value
**My question is where do tell the Nixie dashboard where to go get the probe temp**

**Reply by tagyoureit** [reply]
See this thread for the temp probe setup: https://github.com/tagyoureit/nodejs-poolController/discussions/995#discussioncomment-10004164
